src/librustc/hir/map/definitions.rs

```diff
@@ -314,6 +314,26 @@ impl DefPath {
 
         s
     }
+
+    /// Return filename friendly string of the DefPah without
+    /// the crate-prefix. This method is useful if you don't have
+    /// a TyCtxt available.
+    pub fn to_filename_friendly_no_crate(&self) -> String {
+        let mut s = String::with_capacity(self.data.len() * 16);
+
+        for component in &self.data {
+            if component.disambiguator == 0 {
+                write!(s, ".{}", component.data.as_interned_str()).unwrap();
+            } else {
+                write!(s,
+                       ".{}[{}]",
```

Comment:
Nit: maybe use `-`? I sort of like that the `.` in the current filename allows us to "group" (e.g., `rustc.foo-bar.before.mir` can be split on `.` to get out the various bits of information about what this is a dump of)